### PR TITLE
Default Role/RoleGroup configuration when deserializing

### LINKED ISSUE #281: Configuration::compute_* is not invoked if RoleGroup object doesn't define `config` field
This means that the following object works:

```yaml
apiVersion: imaginary.stackable.tech/v1alpha1
kind: ImaginaryCluster
metadata:
  name: imaginary-cluster
spec:
  role:
    roleGroup:
      config: {}
```

But the following might be missing crucial default values:

```yaml
apiVersion: imaginary.stackable.tech/v1alpha1
kind: ImaginaryCluster
metadata:
  name: imaginary-cluster
spec:
  role:
    roleGroup: {}
```

### PR DESCRIPTION
**Pull request by teozkr**
## Description
Fixes #281

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: src/product_config_utils.rs [hunk 1]
```diff
@@ -418,31 +413,17 @@ where
...
-
     // Properties from the role have the lowest priority, so they are computed and added first...
-    if let Some(CommonConfiguration {
-        config: Some(ref config),
-        ..
-    }) = config
-    {
-        final_properties = config.compute_cli(resource, role_name).unwrap();
-    }
+    let mut final_properties = config.config.compute_cli(resource, role_name).unwrap();
```
Comment (maltesander) [inline src/product_config_utils.rs:422]: It was in before, but i think we should not just unwrap here (and in parse_env_properties, parse_file_properties).
Though we have it under control via framework and operators, come to think of it maybe we want to throw an error if some config properties are wrong, bad combination etc. So i think we should handle this here or at least use unwrap_or_default()?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ All notable changes to this project will be documented in this file.
 ## [Unreleased]
 
 ### Changed
+- BREAKING: Introduced proper error handling for `parse_cli_properties`, `parse_env_properties` and `parse_file_properties` ([#282]).
```
Comment (teozkr) [inline CHANGELOG.md:8]: Isn't `transform_all_roles_to_config` the public part of the API that changed here?

Comment (maltesander) [reply CHANGELOG.md]: True, fixed.

**Comment by teozkr** [comment]
LGTM

Time to give bors the go-ahead?

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request merged**